src/boot/efi/boot.c

```diff
@@ -1493,6 +1497,16 @@ static void config_entry_add_type1(
                         continue;
                 }
 
+                if (streq8(key, "add-on")) {
+                        entry->addons = xrealloc(
+                                entry->addons,
+                                n_addons == 0 ? 0 : (n_addons + 1) * sizeof(uint16_t *),
+                                (n_addons + 2) * sizeof(uint16_t *));
```

Comment:
this expression gives me nightmares. We should have a helper for this somewhere, given this is now used multiple times.

Reply:
Tried something with macros, let me know if you prefer something else in commit fa1afee615